tests/qos/test_tc_to_dscp_mapping.py

```diff
@@ -0,0 +1,504 @@
+"""
+Test cases for testing TC to DSCP mapping for packets in SONiC.
+"""
+
+import json
+import logging
+import random
+import time
+from enum import StrEnum
+from typing import Any
+
+import ptf.packet as scapy
+import ptf.testutils as testutils
+import pytest
+from ptf import mask
+from scapy.all import IP, Ether
+from scapy.layers.l2 import Ether
+from tabulate import tabulate
+
+from tests.common.devices.multi_asic import MultiAsicSonicHost
+from tests.common.dualtor.mux_simulator_control import (
+    toggle_all_simulator_ports_to_rand_selected_tor,
+)  # noqa F401
+from tests.common.fixtures.duthost_utils import dut_qos_maps_module  # noqa F401
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.helpers.dut_utils import check_config_table_presence
+from tests.common.helpers.ptf_tests_helper import (
+    apply_dscp_cfg_setup,
+    apply_dscp_cfg_teardown,
+    downstream_links,
+    get_dut_pair_port_from_ptf_port,
+    get_stream_ptf_ports,
+    select_random_link,
+    upstream_links,
+)  # noqa F401
+from tests.common.platform.processes_utils import wait_critical_processes
+from tests.common.plugins.ptfadapter.ptfadapter import PtfTestAdapter
+from tests.common.utilities import (
+    find_egress_queue,
+    get_dscp_to_queue_value,
+    get_egress_queue_pkt_count_all_prio,
+    get_ipv4_loopback_ip,
+    wait_until,
+)
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [pytest.mark.topology("t0", "t1")]
+
+DEFAULT_ECN = 1
+DEFAULT_PKT_COUNT = 5
+DUMMY_SRC_IP = "8.8.8.8"
+output_table = []
+packet_egressed_success = False
+CONFIG_DB_JSON_PATH: str = "/etc/sonic/config_db.json"
+TC_TO_DSCP_MAP: dict[str, dict[str, str]] = {
+    "AZURE": {
+        "0": "1",
+        "1": "2",
+        "2": "2",
+        "3": "3",
+        "4": "4",
+        "5": "6",
+        "6": "7",
+        "7": "8",
+    }
```

Comment:
Have a negative test where TC is set based on DSCP_TO_TC, but there is no entry for that TC Value in TC_TO_DSCP map. In this case, ingress DSCP should be retained.